### PR TITLE
feat(CounterField): new component

### LINKED ISSUE #189: [Feat]:New Counter Field component
**Description**: The `Counter Field` component allows users to increment or decrement a numeric value using plus and minus buttons or by directly inputting the number. This component is commonly used in scenarios like quantity selection, numeric input, or form fields with a minimum and maximum value.

**Requirements**:

1. **Component Implementation:**

- Implement the `Counter Field` component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and button styles.
- Support configuration options such as minimum, maximum values, and step increments.
- Ensure proper accessibility, including keyboard navigation and ARIA attributes.

2. **Unit Tests:**

- 
- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different configurations (min, max, step).
- Test for edge cases such as invalid input values, exceeding limits, and unsupported step values.

3. **Storybook Stories:**

- Create a Storybook story for the Counter Field component.
- Showcase different configurations (e.g., various min/max values, step sizes).
- Create controls for the props so users can interact with the Counter Field component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=2511-11786&node-type=canvas&t=BW68yyejj2aaHpzs-0)

**Checklist**:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and button styles.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and configuration options.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by HaykKirakosyanSoft**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1129 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-jbyaifasrc.chromatic.com/

### CODE REVIEW

File: src/components/molecules/CounterField/CounterField.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.counterField {
+    $this: &;
+
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    gap: var(--guit-ref-spacing-3xsmall);
+
+    // INPUT CONTAINER - Horizontal layout for buttons and text field
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Please remove all comments from .scss file

File: src/components/molecules/CounterField/CounterField.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    &__inputContainer {
+        display: flex;
+        align-items: center;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: we don't need `align-items: center;` by default input and button has same height.

File: src/components/molecules/CounterField/CounterField.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        gap: var(--guit-ref-spacing-xsmall);
+        width: 100%;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Please remove unnecessary `width: 100%`

File: src/components/molecules/CounterField/CounterField.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    // BUTTON - Decrement and Increment buttons (icon-only secondary fill buttons)
+    &__button {
+        flex-shrink: 0;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Please remove unnecessary style.

File: src/components/molecules/CounterField/CounterField.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    // INPUT - Text field for manual value entry
+    &__input {
+        flex: 0 0 auto;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Why we need `flex: 0 0 auto;`?

File: src/components/molecules/CounterField/CounterField.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    gap: var(--guit-ref-spacing-3xsmall);
+
+    // INPUT CONTAINER - Horizontal layout for buttons and text field
+    &__inputContainer {
+        display: flex;
+        align-items: center;
+        gap: var(--guit-ref-spacing-xsmall);
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss:8]: There is no token for this gap , please clarify with designer

File: src/components/molecules/CounterField/CounterField.scss [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    // INPUT - Text field for manual value entry
+    &__input {
+        flex: 0 0 auto;
+        width: 8rem; // Figma: 80px fixed width
+    }
+
+    // INFO CONTAINER - Helper text area
+    &__infoContainer {
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Please remove unnecessary styles.

File: src/components/molecules/CounterField/CounterField.scss [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        display: flex;
+        width: 100%;
+    }
+
+    // STATE: DISABLED - Prevents all interactions
+    &_disabled {
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: Could we remove these styles? It appears the components' disabled states are managed internally, making these external styles unnecessary.

File: src/components/molecules/CounterField/CounterField.scss [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        pointer-events: none;
+        user-select: none;
+    }
+
+    // STATE: READ-ONLY - Prevents modifications
+    &_readOnly {
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.scss]: this styles unnecessary too.

File: src/components/molecules/CounterField/CounterField.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+const meta: Meta<ICounterFieldProps> = {
+    title: "Molecules/CounterField",
+    component: CounterField,
+    argTypes: {
+        value: args({ control: "number", ...propCategory.states }),
+        defaultValue: args({ control: "number", ...propCategory.states }),
+        disabled: args({ control: "boolean", ...propCategory.states }),
+        readOnly: args({ control: "boolean", ...propCategory.states }),
+
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.stories.tsx]: Please remove unnecessary empty lines.

File: src/components/molecules/CounterField/CounterField.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        required: false
+    }
+};
+
+export default meta;
+
+type Story = StoryObj<ICounterFieldProps>;
+
+const Template: FC<ICounterFieldProps> = ({ ...props }) => (
+    <div style={{ maxWidth: 400 }}>
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.stories.tsx]: We don't need this redundant wrapper , please remove.

File: src/components/molecules/CounterField/CounterField.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    </div>
+);
+
+export const Default: Story = {
+    render: ({ ...props }) => <Template {...props} />,
+    argTypes: { value: args({ control: "false", ...propCategory.states }) }
+};
+
+const ControlledTemplate: FC<ICounterFieldProps> = ({ value, onChange, ...props }) => {
+    const [internalValue, setInternalValue] = React.useState<number>(value ?? 0);
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.stories.tsx]: 1. Please follow the project's import pattern. Destructure React imports (e.g., `import { useState } from 'react'`) instead of using the `React.*` namespace (this is for all files).
2. TypeScript can infer this primitive type. Please remove the explicit `<number>`.

File: src/components/molecules/CounterField/CounterField.test.tsx [hunk 13]
```diff
@@ -0,0 +1,349 @@
+import React from "react";
+import { mount, ReactWrapper } from "enzyme";
+
+import { Minus, Plus } from "@geneui/icons";
+
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.test.tsx:5]: Add `// Components` comment .

File: src/components/molecules/CounterField/CounterField.tsx [hunk 14]
```diff
@@ -0,0 +1,235 @@
+import React, { ChangeEvent, FC, FocusEvent, MouseEvent, useState } from "react";
+import classNames from "classnames";
+
+import { Minus, Plus } from "@geneui/icons";
+
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx:6]: Add `// Components` comment .

File: src/components/molecules/CounterField/CounterField.tsx [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+    /**
+     * Makes the counter field read-only, preventing manual input changes.
+     */
+    readOnly?: boolean;
+
+    // BOUNDARIES & STEP
+    /**
+     * The minimum allowed value.
+     */
+    min?: number;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx:51]: The `CounterField` component currently supports a `min` prop but lacks a `max` prop, which creates an inconsistent API and limits use cases.

File: src/components/molecules/CounterField/CounterField.tsx [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+
+import Button from "@components/atoms/Button";
+import HelperText from "@components/atoms/HelperText";
+import Label from "@components/atoms/Label";
+import TextField from "@components/molecules/TextField";
+
+// Styles
+import "./CounterField.scss";
+
+export interface ICounterFieldProps {
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: All exports must be added at the end of file 
`export { ICounterFieldProps, CounterField as default };`

File: src/components/molecules/CounterField/CounterField.tsx [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+
+    // ACTIONS
+    /**
+     * Fires when the user changes the counter value (via buttons or input).
+     */
+    onChange?: (value: number, event: ChangeEvent<HTMLInputElement> | MouseEvent<HTMLButtonElement>) => void;
+    /**
+     * Fires when the input field loses focus.
+     */
+    onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: Suggestion: Rename `onBlur` and `onFocus` to `onInputBlur` and `onInputFocus` for clarity

The current API is inconsistent:
- `onChange` fires for both input field changes AND button clicks (increment/decrement)
- `onBlur` and `onFocus` only fire for the input field, not the buttons

File: src/components/molecules/CounterField/CounterField.tsx [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+                `counterField_status_${status}`,
+                {
+                    counterField_disabled: disabled,
+                    counterField_readOnly: readOnly
+                },
+                className
+            )}
+            aria-required={required}
+            aria-invalid={status === "error"}
+            {...((disabled || readOnly) && { tabIndex: -1 })}
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: We can remove this line, as the components should handle their own tabIndex internally.

On that note, I don't see a readOnly state for CounterField in the designs. Can we please verify with the designer if this state is needed? If not, we should remove all readOnly logic.

File: src/components/molecules/CounterField/CounterField.tsx [hunk 19]
```diff
@@ -0,0 +1,235 @@
...
+                    readOnly={readOnly}
+                    infoText={infoText}
+                    size={size === "large" ? "medium" : size}
+                />
+            )}
+
+            <div className="counterField__inputContainer">
+                <Button
+                    appearance="secondary"
+                    className="counterField__button counterField__button_decrement"
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: This class name  (`counterField__button_decrement`) needs to be updated to follow our BEM convention. Since this modifier describes the button's action (not a boolean state), it should use a key-value format for example counterField__button_action_decrement.

[read here](https://en.bem.info/methodology/faq/#when-to-create-a-boolean-modifier-and-when-to-create-a-key-value-modifier)

File: src/components/molecules/CounterField/CounterField.tsx [hunk 20]
```diff
@@ -0,0 +1,235 @@
...
+                    size={size}
+                    Icon={Plus}
+                    disabled={disabled || readOnly}
+                    aria-label="Increment value"
+                    onClick={handleIncrement}
+                />
+            </div>
+
+            {helperText && (
+                <div className="counterField__infoContainer">
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: I think this is unnecessary wrapping.

File: src/components/molecules/CounterField/CounterField.tsx [hunk 21]
```diff
@@ -0,0 +1,235 @@
...
+
+            <div className="counterField__inputContainer">
+                <Button
+                    appearance="secondary"
+                    className="counterField__button counterField__button_decrement"
+                    layout="fill"
+                    size={size}
+                    Icon={Minus}
+                    disabled={isDecrementDisabled}
+                    aria-label="Decrement value"
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: Hard-coding any user-facing string, including `aria-label`, makes it impossible to translate.

The Problem: The text "Decrement value" is only in English. A user in Germany or Japan won't understand it, which breaks accessibility for them.

The component should receive its labels as props.

For example.
```
const { decrementAriaLabel = "Decrement value" } = props; 

<button aria-label={decrementAriaLabel}>...</button>
```

File: src/components/molecules/CounterField/CounterField.tsx [hunk 22]
```diff
@@ -0,0 +1,235 @@
...
+     */
+    onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+    /**
+     * Fires when the input field receives focus.
+     */
+    onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * CounterField allows users to increment or decrement a numeric value using buttons or direct input.
```
Comment (GaroGabrielyan) [inline src/components/molecules/CounterField/CounterField.tsx]: The component description should be taken directly from the design.
`The Counter Field component is an input field designed to increment or decrement a numerical value. It typically includes buttons for increasing or decreasing the count and can be configured to accept user input directly.`

**Review by GaroGabrielyan — CHANGES_REQUESTED** [review]
The CounterField component must accept decimal numbers and 'e'.

File: src/components/molecules/CounterField/CounterField.tsx (hunk 20 above)
Comment (HaykKirakosyanSoft) [reply src/components/molecules/CounterField/CounterField.tsx]: Other components using helpertext do it exact same way.

File: src/components/molecules/CounterField/CounterField.scss (hunk 6 above)
Comment (HaykKirakosyanSoft) [reply src/components/molecules/CounterField/CounterField.scss:8]: I saw it on figma, also when you remove this token the gap between label,buttons and input breaks.

**Review by AregSoft — DISMISSED** [review]

**Review by GaroGabrielyan — APPROVED** [review]
